### ISSUE TITLE
Dict comprehensions are not resolved properly inside a screen

### ISSUE THREAD

**Issue by Queex** [renpy/renpy#2267]
This toy script demonstrates the problem:

```
screen testscreen(first_list, another_list):
    default test_val= {x:first_list[i] for i,x in enumerate(another_list)}
    vbox:
        for x in test_val:
            hbox:
                label x
                label str(nested_list[x])

label start:
    $ test_first=[1,2,3]
    $ test_other=["A","B","C"]

    call screen testscreen(test_first, test_other)
```

raises the exception:

```
  File "game/script.rpy", line 2, in execute
    default test_val= {x:first_list[i] for i,x in enumerate(another_list)}
  File "<screen language>", line 2, in <module>
  File "<screen language>", line 2, in <dictcomp>
NameError: global name 'first_list' is not defined

Windows-7-6.1.7601-SP1
Ren'Py 7.3.5.606
ScreenDict 1.0
Sat Jul 04 23:08:03 2020
```

Despite `first_list` being defined within the screen, the list comprehension doesn't appear to find it. I've tried a couple of variations: list comprehensions seem to be evaluated correctly, and the list can be accessed correctly by index otherwise

**Comment by langbakk** [comment]
Fairly certain this is because you use default for the test_var inside the screen, but python statements to assign the lists. Set the lists up as defaults as well, and it'll work, I'm guessing. Or don't use default inside the screen.

**Comment by Queex** [comment]
I first saw the problem when using python statements inside screen. Using `default` for the two variables outside it as well doesn't seem to fix it either.

**Comment by Andykl** [comment]
Hm. Interesting. In both pure python 2.7 and python 3.8 this is exception.
`eval("{i: lvar for i in range(10)}", {}, {"lvar": True})`
I'm not sure is it a python bug.

**Comment by Gouvernathor** [comment]
`eval("(lambda x:{i:x for i in range(10)})(lvar)", {}, {"lvar": True})` works. It is a python glitch, see https://bugs.python.org/issue36300.
Not an expert, but what about merging the local and global dicts in a temporary dict, and give that as the `globals` argument to the `eval` function ?

**Comment by renpytom** [comment]
I believe this will be fixed in  7.4/8.0.
